### ISSUE TITLE
[java] [regression from 6.55] UseTryWithResources false-positive

### ISSUE THREAD

**Issue by kohlschuetter** [pmd/pmd#4614]
**Affects PMD Version:** 

7.0.0-SNAPSHOT

**Rule:**

[https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#usetrywithresources](UseTryWithResources)

**Description:**

**Code Sample demonstrating the issue:**

```java
// both "source()" and "sink()" return Closeable-s
try {
      source().close();
} finally {
      sink().close();
}
```

**Expected outcome:**

PMD reports a violation at line 1, but that's wrong. That's a false positive.
```
Rule:UseTryWithResources Priority:3 Consider using a try-with-resources statement instead of explicitly closing the resource.
```

Changing the clause to a try-with-resources, e.g.
```java
try (Closeable s = source()) {
  // nothing
} finally {
  sink().close();
}
```

will actually cause the following warning instead:

```
Rule:UnusedLocalVariable Priority:3 Avoid unused local variables such as 's'..
```

**Running PMD through:** CLI, Maven

**Comment by oowekyala** [comment]
Hi, you can rename your closeable to something starting with `unused` to suppress the warning of UnusedLocalVariable. It's a bit principled, but I think it's a good practice to be explicit about which variables are unused by design and not by mistake.
```java
try (Closeable unused = source()) {
  // nothing
} finally {
  sink().close();
}
```

**Comment by kohlschuetter** [comment]
Thanks for the tip, @oowekyala, I didn't know that!

However, in order to have no errors, it would have to be

```java
try (Closeable unused = source()) {
  // nothing
} finally {
  try (Closeable unused2 = sink()) {
    // still nothing
  }
}
```

which is pretty awkward, wouldn't you agree?

**Comment by oowekyala** [comment]
Yes it is awkward... Your report only included a violation for `source` so I thought `sink` was ignored for some reason.

Maybe we should not report the pattern `makeCloseable().close()`. In this situation, a TWR doesn't bring anything anyway.

> Also, why didn't the warning show up in 6.55.0?

Type resolution is much better in PMD 7. It's probable that PMD 6 doesn't understand that `source()` and `sink()` return a Closeable.
